AsyncDisplayKit/Details/ASRangeHandlerRender.mm

```diff
@@ -12,64 +12,32 @@
 #import "ASDisplayNode+Subclasses.h"
 #import "ASDisplayNodeInternal.h"
 
-@interface ASDisplayNode (ASRangeHandlerRender)
-
-- (void)display;
-- (void)recursivelyDisplay;
-
-@end
-
-@implementation ASDisplayNode (ASRangeHandlerRender)
-
-- (void)display
-{
-  if (![self __shouldLoadViewOrLayer]) {
-    return;
-  }
-
-  ASDisplayNodeAssertMainThread();
-  ASDisplayNodeAssert(self.nodeLoaded, @"backing store must be loaded before calling -display");
-
-  CALayer *layer = self.layer;
-
-  // rendering a backing store requires a node be laid out
-  [layer setNeedsLayout];
-  [layer layoutIfNeeded];
-
-  if (layer.contents) {
-    return;
-  }
-
-  [layer setNeedsDisplay];
-  [layer displayIfNeeded];
-}
+@implementation ASRangeHandlerRender
 
-- (void)recursivelyDisplay
++ (UIWindow *)workingWindow
 {
-  if (![self __shouldLoadViewOrLayer]) {
-    return;
-  }
-
-  for (ASDisplayNode *node in self.subnodes) {
-    [node recursivelyDisplay];
-  }
-
-  [self display];
+  // we add nodes' views to this invisible window to start async rendering
```

Comment:
just to be *absolutely* safe, can we add an `ASDisplayNodeAssertMainThread()` here?